source/base/hdf5.cc

```diff
@@ -25,6 +25,49 @@ DEAL_II_NAMESPACE_OPEN
 
 namespace HDF5
 {
+  namespace internal
+  {
+    namespace HDF5ObjectImplementation
+    {
+      /**
+       * Abort, should there be an exception being processed (see the error
+       * message).
+       */
+      void
+      check_exception(const std::string &type, const std::string &name)
+      {
+#  ifdef DEAL_II_WITH_MPI
+#    if __cpp_lib_uncaught_exceptions >= 201411
+        // std::uncaught_exception() is deprecated in c++17
+        if (std::uncaught_exceptions() != 0)
+#    else
+        if (std::uncaught_exception() == true)
+#    endif
+          {
+            std::cerr
+              << "---------------------------------------------------------\n"
+              << "HDF5 " + type + " objects call " + name +
+                   " to end access to\n"
+              << "them and release any resources they acquired. This call\n"
+              << "requires MPI synchronization. Since an exception is\n"
+              << "currently uncaught, this synchronization will be skipped\n"
+              << "to avoid a possible deadlock. As a result the HDF5 file\n"
+              << "might be corrupted.\n"
```

Comment:
These last two sentences aren't actually correct any more -- I guess we missed that. Maybe "...currently uncaught, the synchronization would likely deadlock because only the current process is trying to destroy the object. As a consequence, the program will be aborted."

Reply:
Fixed!